### PR TITLE
W3c HTML validation test

### PR DESCRIPTION
**Pull request by borislavpetrovhmcts**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # DIV-1085

## Type of change - implementing properly W3C html validation test, creating mechanism for adding exclusions to the tests, fixing html errors and warning after proper test run 

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
Manually tested.
Automatically tested - unit tests, e2e tests.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: test/validation/htmlTest.js [hunk 1]
```diff
@@ -51,16 +95,34 @@ for (let stepKey in s.steps) {
 
           input: res.text,
 
-          callback: (res) => {
-
+          callback: (error, res) => {
             let warnings = filter(res.messages, (r) => r.type === 'info');
-
-            expect(warnings.length).to.equal(0, JSON.stringify(warnings, null, 2));
+            let filteredWarnings = filter(warnings, (r) =>
```
Comment (mattyod) [inline test/validation/htmlTest.js]: w3cjs is making a network call every time. If we did checks for warnings and errors in the same test run time could be reduced quite a lot.

File: test/validation/htmlTest.js [hunk 2]
```diff
@@ -11,8 +11,44 @@ let agent = {};
...
+  'DesertionAgree',
+  'DesertionDetails',
+  'DesertionDate',
+  'ReasonForDivorce',
+  'SeparationDate',
+  'UnreasonableBehaviour',
+  'NeedHelpWithFees',
+  'WithFees',
+  'GovPayStub'
+].reduce(function (res, key) {
```
[the remark this answers is not in the document]
Comment (borislavpetrovhmcts) [reply test/validation/htmlTest.js]: This is fine

File: test/validation/htmlTest.js (hunk 1 above)
Comment (borislavpetrovhmcts) [reply test/validation/htmlTest.js]: This was implemented according to documentation.

File: test/validation/htmlTest.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mattyod) [reply test/validation/htmlTest.js]: It's harder to reason with in my opinion but if you disagree.

[the remark this answers is not in the document]
Comment (borislavpetrovhmcts) [reply test/validation/htmlTest.js]: This is working fine, I see no reason to refactor it

File: test/validation/htmlTest.js (hunk 1 above)
Comment (mattyod) [reply test/validation/htmlTest.js]: Still makes 2x network calls for no reason.

**Comment by vakepati** [comment]
looks Good

**Comment by vakepati** [comment]
all looks good

**Pull request merged**